src/main/java/io/stargate/sgv2/jsonapi/util/YamlMerger.java

```diff
@@ -0,0 +1,139 @@
+package io.stargate.sgv2.jsonapi.util;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.POJONode;
+import com.fasterxml.jackson.databind.node.ValueNode;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Utility for deep-merging YAML documents with simple, predictable semantics:
+ *
+ * <ul>
+ *   <li>Objects: recursively merged; fields from patch override or extend base
+ *   <li>Arrays: replaced entirely when present in patch
+ *   <li>Scalars (string/number/boolean/null): replaced by patch value
+ * </ul>
+ */
+public final class YamlMerger {
+
+  private final ObjectMapper yamlMapper;
+
+  public YamlMerger() {
+    this.yamlMapper = new ObjectMapper(new YAMLFactory());
```

Comment:
Works, but there's convenience type `YAMLMapper` to use ("this.yamlMapper = new YAMLMapper()")